proxy/http/HttpBodyFactory.cc

```diff
@@ -406,6 +414,12 @@ HttpBodyFactory::fabricate(StrList *acpt_language_list, StrList *acpt_charset_li
     set = determine_set_by_language(acpt_language_list, acpt_charset_list);
   } else if (enable_customizations == 3) {
     set = determine_set_by_host(context);
+  } else if (context->http_return_code == HTTP_STATUS_NO_CONTENT) {
```

Comment:
Could these explicit status checks be replaced by `response_body_precluded` calls?